embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/IntegrationTestRuleExtensions.kt

```diff
@@ -97,8 +97,8 @@ internal fun IntegrationTestRule.Harness.recordSession(
     // end session 30s later by entering background
     overriddenClock.tick(30000)
     activityController?.pause()
-    activityService.onBackground()
     activityController?.stop()
```

Comment:
@fractalwrench is it guaranteed that that activityLifecycleListners will fire before the ProcessLifecycleListeners if both have onStop() listeners registered?  We require the view span to end before the session can be stopped, so we need to ensure there's no race condition between to listener calls so the order can be guaranteed,

Reply:
Interesting question! `registerActivityLifecycleCallbacks` uses an `ArrayList` internally, and the docs for `ProcessLifecycleOwner` state the callback gets added via codegen or reflection. My assumption is that whatever class gets instantiated first will get called first. I'd assume that `EmbraceProcessStateService` gets instantiated first.